src/components/post/post.js

```diff
@@ -31,21 +32,22 @@ export default function Post({ post }) {
 
     const removeBookmark = async () => {
         try {
-            const res = await axios.post(`/api/users/remove-bookmark/${_id}`, {},
-                {
-                    headers: {
-                        authorization: encodedToken
-                    }
-                });
-            setUser({ ...user, bookmarks: res.data.bookmarks })
+            const userRef = doc(db, `users/${user.id}`);
```

Comment:
if the queries feel repetitive put them in a firebaseQueries file and reuse it for better modularity